packages/@ama-mfe/ng-utils/src/navigation/restore-route.pipe.ts

```diff
@@ -0,0 +1,95 @@
+import {
+  inject,
+  Pipe,
+  PipeTransform,
+  SecurityContext,
+} from '@angular/core';
+import {
+  DomSanitizer,
+  type SafeResourceUrl,
+} from '@angular/platform-browser';
+import {
+  ActivatedRoute,
+} from '@angular/router';
+import {
+  RouteMemorizeService,
+} from './route-memorize/route-memorize.service';
+
+/**
+ * Options for restoring a route with optional query parameters and memory channel ID.
+ */
+export interface RestoreRouteOptions {
+  /**
+   * Whether to propagate query parameters from the top window to the module URL.
+   */
+  propagateQueryParams?: boolean;
+
+  /**
+   * Whether to override existing query parameters in the module URL with those from the top window.
+   */
+  overrideQueryParams?: boolean;
+
+  /**
+   * The memory channel ID used to retrieve the memorized route.
+   * If provided, the memorized route associated with this ID will be used.
+   */
+  memoryChannelId?: string;
+}
+
+/**
+ * A pipe that restores a route with optional query parameters and memory channel ID.
+ *
+ * This pipe is used to transform a URL or SafeResourceUrl by appending query parameters
+ * and adjusting the pathname based on the current active route and memorized route.
+ */
+@Pipe({
+  name: 'restoreRoute'
+})
+export class RestoreRoute implements PipeTransform {
+  private readonly activeRoute = inject(ActivatedRoute);
+  private readonly domSanitizer = inject(DomSanitizer);
+  private readonly routeMemorizeService = inject(RouteMemorizeService, { optional: true });
+
+  /**
+   * Transforms the given URL or SafeResourceUrl by appending query parameters and adjusting the pathname.
+   * @param url - The URL or SafeResourceUrl to be transformed.
+   * @param options - Optional parameters to control the transformation. {@see RestoreRouteOptions}
+   * @returns - The transformed SafeResourceUrl or undefined if the input URL is invalid.
+   */
+  public transform(url: string, options?: Partial<RestoreRouteOptions>): string;
+  public transform(url: SafeResourceUrl, options?: Partial<RestoreRouteOptions>): SafeResourceUrl;
+  public transform(url: undefined, options?: Partial<RestoreRouteOptions>): undefined;
+  public transform(url: string | SafeResourceUrl | undefined, options?: Partial<RestoreRouteOptions>): string | SafeResourceUrl | undefined {
+    if (!url) {
+      return undefined;
+    }
+    const urlString = typeof url === 'string'
+      ? url
+      : this.domSanitizer.sanitize(SecurityContext.RESOURCE_URL, url);
```

Comment:
```suggestion
    const urlString = typeof url === 'string'
      ? url
      : this.domSanitizer.sanitize(SecurityContext.RESOURCE_URL, url);
    if (!url) {
      return undefined;
    }
```

to avoid the falling `if` and final `return undefined`